test/function/samples/cycles-pathological-2/_config.js

```diff
@@ -3,9 +3,21 @@ var assert = require( 'assert' );
 module.exports = {
 	description: 'resolves even more pathological cyclical dependencies gracefully',
 	buble: true,
-	options: {
-		onwarn: function ( message ) {
-			assert.ok( /unable to evaluate without/.test( message ) );
+	warnings: [
+		{
+			code: 'CIRCULAR_DEPENDENCY',
+			importer: 'main.js',
+			message: 'Circular dependency: main.js -> b.js -> main.js'
+		},
+		{
+			code: 'CIRCULAR_DEPENDENCY',
+			importer: 'b.js',
+			message: 'Circular dependency: b.js -> d.js -> c.js -> b.js'
+		},
+		{
+			code: 'CIRCULAR_DEPENDENCY',
+			importer: 'main.js',
+			message: 'Circular dependency: main.js -> b.js -> d.js -> c.js -> main.js'
```

Comment:
Ideally we should avoid warning each module in the cycle I think for readability with these messages?

One option would be to filter by unique cycle here, although that may be quite involved. Another option might be to just permit one cycle warning per module (perhaps keying `cycles` by id and then checking each module against this when populating a cycle path, skipping new cycles that have a match of a module here).

Reply:
I think these are three different cycles. The last two share the `b -> d -> c` part of the path, but otherwise they are different.

Reply:
I agree, though, that in real-world code cycles seem to overlap a lot or share a root cause and deduplication would be nice.

Reply:
Ok I see they are fully distinct already so perhaps this won't be necessary. We can always see how it works out in real world cases and if it gets too noisy can then consider such filtering further.

Reply:
Just rechecking https://github.com/rollup/rollup-plugin-commonjs/issues/248 found about this PR which i did overlooked earlier. Great work here, looking forward to full circular references support, though :-)